### ISSUE TITLE
Change transaction gas_budget to be in MIST

### ISSUE THREAD

**Issue by lxfind** [MystenLabs/sui#7731]
Currently gas_budget is in internal gas unit. This is confusing in several ways:
1. A user only cares about the total MIST/SUI spent in a transaction, not internal gas units.
2. When we translate this to SUI to check gas balance, we have to use the max of computation gas price and storage gas price, which is strange and could lead to confusing results.

The fix is to change the meaning of the gas_budget field in transaction to mean total MIST that can be spent by the transaction.
This will require a restructuring of the SuiGasStatus so that we track the remaining budget in MIST, and translate that into internal gas unit whenever we need to run Move.

**Comment by bwann52** [comment]
I agree that most users only care about total gas (measured in MIST/SUI) paid in a transaction;

But I can also see the benefit of showing granular calculation breakdown of price (or RGB) x gas units = gas fee
- more clarity on how gas fee is calculated
- better communicate our gas tokenomics, how RGP x gas units = gas fee paid
- could be insightful later on when we start to utilize congestion control (gas budget bidding for priority)

What exactly, is the current and future formula we will use to derive gas needed to be paid in transaction?

**Comment by dariorussi** [comment]
the current plan is to implement this and move away from gas unit. 
There are several reasons for that, including making gas cost in coarse buckets.
Not clear there is much benefit in giving users details that may be even more difficult to relate to in time.
